nodeadm/api/v1alpha1/nodeconfig_types.go

```diff
@@ -29,10 +29,11 @@ type NodeConfigList struct {
 }
 
 type NodeConfigSpec struct {
-	Cluster    ClusterDetails    `json:"cluster,omitempty"`
-	Containerd ContainerdOptions `json:"containerd,omitempty"`
-	Instance   InstanceOptions   `json:"instance,omitempty"`
-	Kubelet    KubeletOptions    `json:"kubelet,omitempty"`
+	Cluster      ClusterDetails    `json:"cluster,omitempty"`
+	Containerd   ContainerdOptions `json:"containerd,omitempty"`
+	Instance     InstanceOptions   `json:"instance,omitempty"`
+	Kubelet      KubeletOptions    `json:"kubelet,omitempty"`
+	FeatureGates map[string]bool   `json:"featureGates,omitempty"`
```

Comment:
needs a doc description?

Reply:
Yes, doc for `FeatureGates` will be easy, but I think we will also need a one for the configurable features and another guidance doc on how to use this flag for node name, I'm still working on test and writing up that doc.